vendor/bat-native-ads/src/bat/ads/internal/database/tables/dayparts_database_table.cc

```diff
@@ -0,0 +1,170 @@
+#include "bat/ads/internal/frequency_capping/exclusion_rules/daypart_frequency_cap.h"
+#include "bat/ads/internal/logging.h"
+
+namespace ads {
+namespace database {
+namespace table {
+
+using std::placeholders::_1;
+
+namespace {
+const char kTableName[] = "dayparts";
+}  // namespace
+
+Dayparts::Dayparts(
+    AdsImpl* ads)
+    : ads_(ads) {
+  DCHECK(ads_);
+}
+
+Dayparts::~Dayparts() = default;
+
+void Dayparts::InsertOrUpdate(
+    DBTransaction* transaction,
+    const CreativeAdList& creative_ads) {
+  DCHECK(transaction);
+
+  if (creative_ads.empty()) {
+    return;
+  }
+
+  DBCommandPtr command = DBCommand::New();
+  command->type = DBCommand::Type::RUN;
+  command->command = BuildInsertOrUpdateQuery(command.get(),
+      creative_ads);
+
+  transaction->commands.push_back(std::move(command));
+}
+
+void Dayparts::Delete(
+    ResultCallback callback) {
+  DBTransactionPtr transaction = DBTransaction::New();
+
+  util::Delete(transaction.get(), get_table_name());
+
+  ads_->get_ads_client()->RunDBTransaction(std::move(transaction),
+      std::bind(&OnResultCallback, _1, callback));
+}
+
+std::string Dayparts::get_table_name() const {
+  return kTableName;
+}
+
+void Dayparts::Migrate(
+    DBTransaction* transaction,
+    const int to_version) {
+  DCHECK(transaction);
+
+  switch (to_version) {
+    case 3: {
+      MigrateToV3(transaction);
+      break;
+    }
+
+    default: {
+      break;
+    }
+  }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+int Dayparts::BindParameters(
+    DBCommand* command,
+    const CreativeAdList& creative_ads) {
+  DCHECK(command);
+
+  int count = 0;
+  int index = 0;
+
+  for (const auto& creative_ad : creative_ads) {
+    for (const auto& daypart : creative_ad.dayparts) {
+      std::cout << "albert inserting into database: ";
+      BindString(command, index++,
+          creative_ad.campaign_id);
+      BindString(command, index++, daypart.days_of_week);
+      BindInt(command, index++, daypart.start_minute);
+      BindInt(command, index++, daypart.end_minute);
+
+      count++;
+    }
+  }
+
+  return count;
+}
+
+std::string Dayparts::BuildInsertOrUpdateQuery(
+    DBCommand* command,
+    const CreativeAdList& creative_ads) {
+  const int count = BindParameters(command, creative_ads);
+
+  return base::StringPrintf(
+      "INSERT OR REPLACE INTO %s "
+          "(campaign_id, "
+          "days_of_week, "
+          "start_minute, "
+          "end_minute) VALUES %s",
+      get_table_name().c_str(),
+      BuildBindingParameterPlaceholders(4, count).c_str());
+}
+
+void Dayparts::CreateTableV3(
+    DBTransaction* transaction) {
+  DCHECK(transaction);
+
+  const std::string query = base::StringPrintf(
+      "CREATE TABLE %s "
+          "(campaign_id TEXT NOT NULL, "
+          "days_of_week TEXT NOT NULL, "
+          "start_minute INT NOT NULL, "
+          "end_minute INT NOT NULL, "
+          "CONSTRAINT fk_campaign_id "
```

Comment:
Delete cascade will not work here, please remove as table is rebuilt for catalog updates